pkg/pipeline/crd.go

```diff
@@ -66,6 +67,12 @@ func (cg *CRDGenerator) Generate(cfg *config.Resource) (string, error) {
 	for _, omit := range cfg.ExternalName.OmittedFields {
 		delete(cfg.TerraformResource.Schema, omit)
 	}
+
+	cfg.TerraformResource.Schema["id"] = &schema.Schema{
```

Comment:
As we discussed in sync, `id` field is [integral part](https://learn.hashicorp.com/tutorials/terraform/provider-create#implement-create) (search for `SetId`) of the state because Terraform assumes that resource does not exist if `id` is not set. So, it should be safe to always add it to the schema.

Reply:
I agree. One point to keep in mind is that we have some further assumptions like in the schema, it's computed.

Reply:
At that level, I think we can add a nil guard, like `if cr.Status.AtProvider.ID == nil { return "" }` in `GetID() string`. My point was that we are sure that field is reserved for `id`, so we don't have to check whether it comes with the schema.